### ISSUE TITLE
index Folio identifier(s), e.g. HRID to Argo prod index (in addition to indexing Symphony catkeys)

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/dor_indexing_app#930]
Prereq:

- [ ] decide on prefix, which may influence the field name.  See https://stanfordlib.slack.com/archives/C04EZLBMLQN/p1675977304131999 as well as comment below.

See https://github.com/sul-dlss/dor_indexing_app/blob/main/app/indexers/identity_metadata_indexer.rb#L16-L20 and `catkey` and `prefixed_identifier` methods.

New Fields to be added:
      `folio_instance_hrid_ssim`- analogous to 'catkey_id_ssim' => [catkey].compact,

Fields to change:
  - add folio hrid value prefixed with "folio" to prefixed_identifiers so it goes in indexifier_xxsim fields
  - add folio hrid value to  'dor_id_tesim' => [source_id_value, barcode, catkey].compact (will make it searched)

To test:

With this change deployed to dor-indexing-app stage, try re-indexing a record with a Folio instance HRID in its cocina and see if it:
- turns up in Solr search results when you search Solr argo-stage with the Folio id
- turns up in Argo search results when you search on the Folio id

You might want to manually set a DSA-stage cocina folio id to something unique like zzzsngbt or whatever.  (and then fix it -- maybe unset it and reindex).

Partially blocked by sul-dlss/cocina-models#561, and an instance of DSA using cocina-models with that PR (either as a branch or bundle updated after cocina-models gem release with that change).  These changes should go out on Monday Feb 13 with dep updates by First Responder.

**Comment by ndushay** [comment]
@andrewjbtw @arcadiafalcone what we call the field in the Solr index is a question.   "catkey" is taken and is in all our data.  I suggest that "catalog_record_id" may not be the best prefix to use in the `identiier_xxsim` fields, which I believe appear in the show view and the search results.   What would you like?  I'm marking this ticket blocked until that's decided

**Comment by arcadiafalcone** [comment]
I think this is an area in which more specificity may be better - how about `folio_instance_hrid`?

**Comment by arcadiafalcone** [comment]
Andrew concurs, so let's go with `folio_instance_hrid`.

**Comment by ndushay** [comment]
closing this in favor of newly written issue #935
